src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -102,17 +118,24 @@ private async Task<int> StartMonitor()
                 }
                 providerString = sb.ToString();
             }
-
-            var configuration = new SessionConfiguration(
-                circularBufferSizeMB: 1000,
-                outputPath: outputPath,
-                providers: Trace.Extensions.ToProviders(providerString));
-
-            sessionId = EventPipeClient.StartTracingToFile(_processId, configuration);
-
-            // Write the config file contents
-            _console.Out.WriteLine("Tracing has started. Press Ctrl-C to stop.");
-            await Task.Delay(int.MaxValue, _ct);
+            Task monitorTask = new Task(() => {
+                var configuration = new SessionConfiguration(
+                    circularBufferSizeMB: 1000,
+                    outputPath: "",
+                    providers: Trace.Extensions.ToProviders(providerString));
+
+                var binaryReader = EventPipeClient.CollectTracing(_processId, configuration, out _sessionId);
+                EventPipeEventSource source = new EventPipeEventSource(binaryReader);
+                writer.InitializeDisplay();
+                source.Dynamic.All += Dynamic_All;
+                source.Process();
+            });
+
+            monitorTask.Start();
+            monitorTask.Wait();
+            EventPipeClient.StopTracing(_processId, _sessionId);
+
+            await Task.FromResult(0);
```

Comment:
This looks like it was added to squelch the warning that an async method had no awaits. You could remove it after adding the await above.